homeassistant/components/light/flux_led.py

```diff
@@ -222,27 +222,34 @@ def turn_on(self, **kwargs):
         effect = kwargs.get(ATTR_EFFECT)
         white = kwargs.get(ATTR_WHITE_VALUE)
 
-        # color change only
-        if rgb is not None:
-            self._bulb.setRgb(*tuple(rgb), brightness=self.brightness)
+        # show warning if effect set with rgb, brightness, or white level
+        if effect and (brightness or white or rgb):
+            _LOGGER.warning("RGB, brightness, and white level are ignored when"
+                            " an effect is specified for a flux bulb.")
 
-        # brightness change only
-        elif brightness is not None:
-            (red, green, blue) = self._bulb.getRgb()
-            self._bulb.setRgb(red, green, blue, brightness=brightness)
+        # preserve current brightness on color/white level change
+        if brightness is None:
+            brightness = self.brightness
+
+        # preserve color on brightness/white level change
+        if rgb is None:
+            rgb = self._bulb.getRgb()
```

Comment:
Move the brightness and rgb assignment to after the `effect` conditionals to avoid this logic. (Not sure if `getRgb` requires I/O, so it'd be nice to avoid that in case it does).

After that, I think this is good to go

Reply:
Updated.  Thanks for the suggestions.